addons/mail/static/src/core/common/message.js

```diff
@@ -423,11 +422,9 @@ export class Message extends Component {
         }
         if (ev.target.closest(".o_mail_redirect")) {
             ev.preventDefault();
-            const partnerId = Number(ev.target.dataset.oeId);
-            if (user.partnerId !== partnerId) {
-                this.threadService.openChat({ partnerId });
+            if (this.store.self.type === "guest") {
+                return;
             }
-            return;
```

Comment:
Only Public and Portal users will be using this code.  Therefore, just return here without any checks, as Public/Portal users won't be able to access the form views of the Partner.